api/enki.py

```diff
@@ -249,17 +294,136 @@ def checkout(self, patron, pin, licensepool, internal_format):
         )
         return loan
 
-    def get_loan(barcode, pin, book_id):
-        self.log.debug ("Sending checkout request for %d" % book_id)
-        now = datetime.datetime.utcnow()
+    def loan_request(self, barcode, pin, book_id):
+        self.log.debug ("Sending checkout request for %s" % book_id)
         url = str(self.base_url) + str(self.user_endpoint)
         args = dict()
         args['method'] = "getSELink"
-        args['username'] = barcide
+        args['username'] = barcode
         args['password'] = pin
         args['lib'] = self.library_id
         args['id'] = book_id
+
+        response = self.request(url, method='get', params=args)
+        return response
+
+    def fulfill(self, patron, pin, licensepool, internal_format):
+        book_id = licensepool.identifier.identifier
+        response = self.loan_request(patron.authorization_identifier, pin, book_id)
+        if response.status_code != 200:
+            raise CannotFulfill(response.status_code)
+        result = json.loads(response.content)['result']
+        if not result['success']:
+            message = result['message']
+            if "That record is already checked out to you" in message:
+                # this might be okay
+                pass
```

Comment:
This isn't okay. Here you know you have a loan and you're asking for a copy of the book. If the error is "the book is already checked out" that indicates the server thinks you're trying to create another loan. Hopefully this is copy-and-paste from the checkout() method that can be removed.

Reply:
This error message is, I believe, a relic of how I was doing things before. The request we're sending to Enki will never return this message. On the first request, the book will be checked out if it isn't already. On subsequent requests, it will still return that book's information without creating any new loan records.